metamorph/store/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,461 @@
+package dynamodb
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/bitcoin-sv/arc/metamorph/metamorph_api"
+	"github.com/bitcoin-sv/arc/metamorph/store"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/ordishs/gocore"
+)
+
+type DynamoDB struct {
+	dynamoCli *dynamodb.Client
+}
+
+const ISO8601 = "2006-01-02T15:04:05.999Z"
+
+func New() (store.MetamorphStore, error) {
+	// Using the SDK's default configuration, loading additional config
+	// and credentials values from the environment variables, shared
+	// credentials, and shared configuration files
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(os.Getenv("AWS_REGION")))
+	if err != nil {
+		return &DynamoDB{}, err
+	}
+
+	// create dynamodb client
+	dynamodbClient := &DynamoDB{dynamoCli: dynamodb.NewFromConfig(cfg)}
+
+	// create table if not exists
+	exists, err := dynamodbClient.TableExists("transactions")
+	if err != nil {
+		return dynamodbClient, err
+	} else if !exists {
+		err = dynamodbClient.CreateTransactionsTable()
+		if err != nil {
+			return dynamodbClient, err
+		}
+	}
+
+	// create table if not exists
+	exists, err = dynamodbClient.TableExists("blocks")
+	if err != nil {
+		return dynamodbClient, err
+	} else if !exists {
+		err = dynamodbClient.CreateBlocksTable()
+		if err != nil {
+			return dynamodbClient, err
+		}
+	}
+
+	// Using the Config value, create the DynamoDB client
+	return dynamodbClient, nil
+}
+
+func (ddb *DynamoDB) TableExists(tableName string) (bool, error) {
+	// Build the request with its input parameters
+	resp, err := ddb.dynamoCli.ListTables(context.TODO(), &dynamodb.ListTablesInput{})
+	if err != nil {
+		return false, err
+	}
+
+	for _, tn := range resp.TableNames {
+		if tn == tableName {
+
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// CreateTransactionsTable creates a DynamoDB table for storing metamorph user transactions
+func (ddb *DynamoDB) CreateTransactionsTable() error {
+	// construct primary key and create table
+	if _, err := ddb.dynamoCli.CreateTable(context.TODO(), &dynamodb.CreateTableInput{
```

Comment:
This if clause is huge.
I personally find this nicer, but it is a matter of taste

```
err := ddb.dynamoCli.CreateTable(context.TODO(), &dynamodb.CreateTableInput{...}
if err != nil { ... }
```